kyaml/fn/framework/command/command.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Build returns a cobra.Command to run a function.
+//
+// The cobra.Command will read the input from STDIN, invoke the provided processor,
+// and then write the output to STDOUT.
+//
+// The cobra.Command will have a boolean `--stack` flag to print stack traces on failure.
+//
+// The cobra.Command will also have a "gen" subcommand that can be used to generate
+// a Dockerfile to build the function into a container image.
+//
+//		go run main.go gen DIR/
+//
+// By default, invoking the returned cobra.Command with arguments triggers "standalone" mode.
+// In this mode:
+// - The first argument must be the name of a file containing the FunctionConfig.
+// - The remaining arguments must be filenames containing input resources for ResourceList.Items.
+// - The argument "-", if present, will cause resources to be read from STDIN as well.
+// The output will be a raw stream of resources (not wrapped in a List type).
+// Example usage: `cat input1.yaml | go run main.go config.yaml input2.yaml input3.yaml -`
+//
+// If `disableStandalone` is set to true, all arguments are ignored, and STDIN must contain
+// a Kubernetes List type. To pass a function config in this mode, use a ResourceList as the input.
+// The output will be of the same type as the input (e.g. ResourceList).
+// Example usage: `cat resource_list.yaml | go run main.go`
+//
+// By default, any error returned by the ResourceListProcessor will be printed to STDERR.
+// Set noPrintError to true to suppress this.
+func Build(p framework.ResourceListProcessor, disableStandalone, noPrintError bool) *cobra.Command {
```

Comment:
Is there any reason why the booleans are both negations?
I feel like it would be clearer to have `standalone` and `printError`

If we want to make this API harder to misuse, I would propose having a type for the mode

```go
type BuildMode byte

const (
    Standalone BuildMode = iota
    NonStandalone // or whatever this should be called
)
```

That way, the calls would be `command.Build(p, command.Standalone, true)` which makes it a bit easier on the reader